streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -744,6 +820,34 @@ private void removeStandbyTasks() {
         standbyRecords.clear();
     }
 
+    private void removeSuspendedTasks() {
+        log.info("{} Removing all suspended tasks [{}]", logPrefix, suspendedTasks.keySet());
+        try {
+            // Close task and state manager
+            for (final AbstractTask task : suspendedTasks.values()) {
+                task.close();
```

Comment:
Do we want to flush the producer after this call? i.e., `task.close` could result in more data being produced.